pkg/apis/core/v1/defaults.go

```diff
@@ -137,32 +136,36 @@ func SetDefaults_Service(obj *v1.Service) {
 		obj.Spec.ExternalTrafficPolicy = v1.ServiceExternalTrafficPolicyTypeCluster
 	}
 
-	// if dualstack feature gate is on then we need to default
-	// Spec.IPFamily correctly. This is to cover the case
-	// when an existing cluster have been converted to dualstack
-	// i.e. it already contain services with Spec.IPFamily==nil
-	if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) &&
-		obj.Spec.Type != v1.ServiceTypeExternalName &&
-		obj.Spec.ClusterIP != "" && /*has an ip already set*/
-		obj.Spec.ClusterIP != "None" && /* not converting from ExternalName to other */
-		obj.Spec.IPFamily == nil /* family was not previously set */ {
+	if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
```

Comment:
Those are straight forward defaulting (no dependency on how the cluster is configured). My understanding they this *is* where this belongs.

Reply:
It's arguable - yes it CAN go here, but *all* the other logic is far away in a different package, so putting them together seems like a decent tradeoff.

Reply:
i still believe this will complicate things, not simplify things
- This first defaulting/validation call is called before webhooks. Currently webhooks only need to work with nil policies. They don't have to do if `len(families) == 2 and policy == nil`, if we changed this they have to deal with more cases.
- The first validation also specifically asserts 2 families then must be require || prefer dualstack (we will have to remove that) and move it inside rest.

Reply:
I am serving another goal - trying to get rid of defaults.go in almost all cases and move to declarative defaults.l (there's a KEP).  Since this is not trivial it would not be viable as declarative (at least in first phases).  Nor is this likely to change across API versions. That just makes me wonder if there should be another REST hook before mutating admission.  But there isn't (yet) so it's probably sufficient to have this as-is and just commented. here and in rest so a reader will know the other exists.

Reply:
leaving for followup